flaskServer/main.py

```diff
@@ -0,0 +1,217 @@
+
+log_file_name = "server.log"
+log_file_path = os.path.join(LOGS_DIRECTORY, log_file_name)
+
+handler = logging.handlers.TimedRotatingFileHandler(log_file_path, when="D", interval=1, backupCount=5)
+handler.setFormatter(logging.Formatter("[%(levelname)s] %(asctime)s - %(message)s"))
+app.logger.addHandler(handler)
+app.logger.setLevel(logging.INFO)
+
+
+def allowed_file(filename: str) -> bool:
+    """
+    Check if the given filename has a valid extension.
+
+    Args:
+        filename (str): The name of the file.
+
+    Returns:
+        bool: True if the file has a valid extension, False otherwise.
+    """
+    return '.' in filename and \
+           filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
+
+
+@dataclass
+class Status:
+    status: str
+    filename: str
+    timestamp: str
+    explanation: dict
+
+    def is_done(self) -> bool:
+        """
+        Check if the status is 'done'.
+
+        Returns:
+            bool: True if the status is 'done', False otherwise.
+        """
+        return self.status == 'done'
+
+
+@app.route('/upload', methods=['POST'])
+def upload_file() -> jsonify:
+    """
+    Handle the file upload request.
+
+    Returns:
+        jsonify: JSON response with the unique ID (uid) of the uploaded file.
+    """
+
+    if 'file' not in request.files:
+        app.logger.info('No file part')
+        return jsonify(uid="0"), 404
+
+    file = request.files['file']
+
+    if file.filename == '':
+        app.logger.info('Empty filename')
+        return jsonify(uid="0"), 404
+
+    if file and allowed_file(file.filename):
+        unique_id = str(uuid.uuid4())
+        print('allowed file received')
+        filename = f'{unique_id}_{datetime.datetime.now()}_{secure_filename(file.filename)}'
+        file.save(os.path.join(app.config['UPLOAD_DIRECTORY'], filename))
+        app.logger.info('File uploaded: %s', filename)
+        return jsonify(uid=unique_id)
+
+
+@app.route("/status/<uid>", methods=['GET'])
+def get_status(uid: str) -> jsonify:
+    """
+    Get the status of a file by its unique ID (uid).
+
+    Args:
+        uid (str): The unique ID of the file.
+
+    Returns:
+        jsonify: JSON response with the status, filename, timestamp, and explanation (if available).
+    """
+    file_name, uid_number, time_stamp, explanation = None, None, None, None
+    response_status = 200
+
+    file_path_upload = glob.glob(os.path.join(UPLOAD_DIRECTORY, f'{uid}*'))
+    file_path_output = glob.glob(os.path.join(OUTPUTS_DIRECTORY, f'{uid}*'))
+
+    if file_path_upload:
+        split_name = os.path.basename(file_path_upload[0]).split('_')
+        uid_number, time_stamp, *file_name_parts = split_name
+        file_name = '_'.join(file_name_parts)
+        if file_path_output:
+            status = 'done'
+
+            try:
+                with open(file_path_output[0], 'r') as f:
+                    explanation = json.load(f)
+            except Exception as e:
```

Comment:
The exception that would be caught would be a specific one.
Better be explicit than implicit.